app/server/appsmith-plugins/restApiPlugin/src/main/java/com/external/helpers/DatasourceValidator.java

```diff
@@ -6,9 +6,18 @@
 
 import java.util.HashSet;
 import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class DatasourceValidator {
 
+    private static final String URL_REGEX =
+            "^(https?://" +
+                    "(%[0-9A-Fa-f]{2}|[-()_.!~*';/?:@&=+$,A-Za-z0-9])+)" +
+                    "([).!';/?:,][[:blank:]])?$";
```

Comment:
What is the last group for? Doesn't look like it's needed, and I can't find the `[[:blank:]]` syntax on the [Pattern class either](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html)?

```suggestion
            "^https?://" +
                    "(%[0-9A-Fa-f]{2}|[-()_.!~*';/?:@&=+$,A-Za-z0-9])+$"
```

This seems to work just as well no?